### PR TITLE
Issue #283: take_while operator implementation

### PR DESCRIPTION
**Pull request by Ravirael**
This PR contains `take_while` operator implementation with tests. I'm not sure about coding standards and some of design decisions so I based the implementation mainly on `take` operator. Testes also are almost copy of `take` operator tests. Solves issue #283.

### CODE REVIEW

**Comment by msotclas** [comment]
Hi __@Ravirael__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla2.msopentech.com.
    </span>

TTYL, MSOTBOT;

**Comment by msotclas** [comment]
__@Ravirael__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, MSOTBOT;

**Comment by kirkshoop** [comment]
This is great! Thank you for the PR!

We are in the midst of a transition to a new pattern for operator implementation. `take()` had not been updated yet, so `take_while()` is also in the old style. I will merge this happily, but please take a look at #280 to see an example of updating the pattern.

If you would like to update `take_while()` in another PR, please do. Otherwise we will pick it up as we go through the existing operators.

File: Rx/v2/src/rxcpp/rx-observable.hpp [hunk 1]
```diff
@@ -2940,6 +2940,24 @@ class observable
...
     }
 
+    /*! For the first items fulfilling the predicate from this observable emit them from the new observable that is returned.
+
+    \tparam Predicate  the type of the predicate
+
+    \param t  the predicate
+
+    \return  An observable that emits only the first items emitted by the source Observable fulfilling the predicate, or all of the items from the source observable if the predicate never returns false
+*/
```
Comment (kirkshoop) [inline Rx/v2/src/rxcpp/rx-observable.hpp]: A snippet needs to be added to the docs in another PR.

Thanks!

example of the `take_until()` snippet [definition](https://github.com/Reactive-Extensions/RxCpp/blob/master/Rx/v2/examples/doxygen/take_until.cpp) & [build reference](https://github.com/Reactive-Extensions/RxCpp/blob/master/projects/doxygen/CMakeLists.txt#L102) & doc reference
```
/*
        \sample
        \snippet take_until.cpp threaded take_until time sample
        \snippet output.txt threaded take_until time sample
*/
```

**Comment by Ravirael** [comment]
Ups... RxCpp code is a bit overwhelming with varadic templates and other kinds of magic everywhere. I will see if I can refactor this, though it would be easier having `take()` already refactored.
As far as docs are concerned I will try update them.

Thank you for the merge and review!

**Comment by kirkshoop** [comment]
Your PR is great! Thank you for taking the time to contribute. I am sure that you will feel more comfortable with the code soon :)
